### PR TITLE
Implement clustering

### PR DESCRIPTION
**Pull request by mbarbar**
This should be the final PR. The `::cluster` method is large, eventually I can shrink it after experiments get going, and I will remove the dependency on `BVDataPTAImpl`.

### CODE REVIEW

File: lib/WPA/Andersen.cpp [hunk 1]
```diff
@@ -187,6 +190,18 @@ void Andersen::initialize()
  */
 void Andersen::finalize()
 {
+    // TODO: check -stat too.
```
Comment (yuleisui) [inline lib/WPA/Andersen.cpp:193]: What's these few lines for in the finalize method of Andersen?

File: lib/WPA/Andersen.cpp [hunk 2]
```diff
@@ -811,6 +826,25 @@ void Andersen::updateNodeRepAndSubs(NodeID nodeId, NodeID newRepId)
     consCG->resetSubs(nodeId);
 }
 
+void Andersen::cluster(void) const
+{
+    assert(Options::MaxFieldLimit == 0 && "Andersen::cluster: clustering for Andersen's is currently only supported in field-insesnsitive analysis");
+    Steensgaard *steens = Steensgaard::createSteensgaard(pag);
```
Comment (yuleisui) [inline lib/WPA/Andersen.cpp:832]: Why we need a steensgaard here?

Comment (mbarbar) [reply lib/WPA/Andersen.cpp:832]: We need data to perform clustering.

File: lib/WPA/Andersen.cpp (hunk 1 above)
Comment (mbarbar) [reply lib/WPA/Andersen.cpp:193]: This is to print the statistics of how effective the clustering actually was (how many words needed).

File: lib/WPA/Andersen.cpp (hunk 2 above)
Comment (yuleisui) [reply lib/WPA/Andersen.cpp:832]: I see. This is to use steensgaard to boost andersen?

Comment (mbarbar) [reply lib/WPA/Andersen.cpp:832]: Exactly

**Pull request merged**